lib/jobs/OperationStatusPollerJob.js

```diff
@@ -41,63 +54,71 @@ class OperationStatusPollerJob extends BaseJob {
     const operationName = job.attrs.data.operation;
     const backupGuid = job.attrs.data.operation_response.backup_guid;
     const boshDirectorName = job.attrs.data.bosh_director;
-    try {
-      Promise.try(() => {
-          if (operationName === 'backup') {
-            return this
-              .getBrokerClient()
-              .getDeploymentBackupStatus(deploymentName, operationResp.token, boshDirectorName);
-          } else if (operationName === 'restore') {
-            return this
-              .getBrokerClient()
-              .getDeploymentRestoreStatus(deploymentName, operationResp.token, boshDirectorName);
-          } else {
-            this.runFailed(new errors.BadRequest(`${operationName} NOT supported for job ${CONST.JOB.OPERATION_STATUS_POLLER}`), {}, job, done);
-          }
-        })
-        .then(operationStatusResponse => {
-          if (isFinished(operationStatusResponse.state)) {
-            ScheduleManager.cancelSchedule(`${deploymentName}_${operationName}_${backupGuid}`, CONST.JOB.OPERATION_STATUS_POLLER)
+
+    return Promise.try(() => {
+        if (operationName === 'backup') {
+          return this
+            .getBrokerClient()
+            .getDeploymentBackupStatus(deploymentName, operationResp.token, boshDirectorName);
+        } else if (operationName === 'restore') {
+          return this
+            .getBrokerClient()
+            .getDeploymentRestoreStatus(deploymentName, operationResp.token, boshDirectorName);
+        } else {
+          throw new errors.BadRequest(`Operation ${operationName} not supported by status poller.`);
+        }
+      })
+      .then(operationStatusResponse => {
+        if (isFinished(operationStatusResponse.state)) {
+          return ScheduleManager.cancelSchedule(`${deploymentName}_${operationName}_${backupGuid}`, CONST.JOB.OPERATION_STATUS_POLLER)
+            .then(() => {
+              logger.info(`Deployment ${deploymentName} ${operationName} for backup guid ${backupGuid} completed -`, operationStatusResponse);
+              return operationStatusResponse;
```

Comment:
it would be a good idea to set the attrib `jobCancelled:true` in this case & can set this default value to `false`

Reply:
done